python/lsst/ts/salobj/sal_info.py

```diff
@@ -571,71 +617,32 @@ async def _run_kafka(self) -> None:
-            ) as connector, aiohttp.ClientSession(connector=connector) as session:
-                registry = RegistryApi(url=self.schema_registry_url, session=session)
-                self._deserializer = Deserializer(registry=registry)
-
-                # A list of Kafka topic names for creating the consumer.
-                read_topic_kafka_names: typing.List[str] = []
-                # A list of TopicInfo for read topics for which historical
-                # data is wanted.
-                read_history_topic_infos: typing.List[TopicInfo] = []
-                for read_topic in self._read_topics.values():
-                    avro_schema = read_topic.topic_info.make_avro_schema()
-                    await registry.register_schema(
-                        schema=avro_schema, subject=read_topic.topic_info.avro_subject
-                    )
-                    read_topic_kafka_names.append(read_topic.topic_info.kafka_name)
-                    if read_topic.max_history > 0:
-                        read_history_topic_infos.append(read_topic.topic_info)
-
-                for write_topic in self._write_topics.values():
-                    avro_schema = write_topic.topic_info.make_avro_schema()
-                    self._serializers[
-                        write_topic.attr_name
-                    ] = await Serializer.register(
-                        registry=registry,
-                        schema=avro_schema,
-                        subject=write_topic.topic_info.avro_subject,
-                    )
-
-                loop = asyncio.get_running_loop()
-                with ThreadPoolExecutor(max_workers=1) as executor:
-                    loop.run_in_executor(
-                        executor=executor, func=self._blocking_create_topics
-                    )
-
-                self._producer = AIOKafkaProducer(
-                    bootstrap_servers=self.kafka_broker_addr,
-                    acks=PRODUCER_WAIT_ACKS,
-                )
-                await self._producer.start()
-                if self.write_only:
-                    if not self.start_task.done():
-                        self.start_task.set_result(None)
-                    await asyncio.Future()
-                else:
-                    self._consumer = AIOKafkaConsumer(
-                        *read_topic_kafka_names,
-                        bootstrap_servers=self.kafka_broker_addr,
-                    )
-                    await self._consumer.start()
-                    self._read_loop_task = asyncio.create_task(self._read_loop())
-                    await self._read_loop_task
+            # Create Kafka topics, serializers, and deserializers.
+            # Set self._serializers_and_contexts and
+            # self._deserializers_and_contexts.
+            await self.loop.run_in_executor(self.pool, func=self._blocking_setup_kafka)
+
+            if not self._read_topics:
+                # Note: if there are read topics,
```

Comment:
I feel like this note should either be before the if statement or in the else statement. Because it is referring to a condition that is not the one in this block.

Reply:
Agreed. I'll add code comments to both the if and else, to make this clearer.